ctapipe/reco/sklearn.py

```diff
@@ -507,6 +501,325 @@ def predict_table(self, key, table: Table) -> Table:
+        # and make `self.load_path` available
+        Component.__init__(self, **kwargs)
+
+        if self.load_path is None:
+            if self.norm_cls is None or self.sign_cls is None:
+                raise TraitError(
+                    "Must provide `norm_cls` and `sign_cls` if not loading from file"
+                )
+
+            if subarray is None:
+                raise TypeError(
+                    "__init__() missing 1 required positional argument: 'subarray'"
+                )
+
+            super().__init__(subarray, **kwargs)
+            self.quality_query = MLQualityQuery(parent=self)
+            self.feature_generator = FeatureGenerator(parent=self)
+
+            # to verify settings
+            self._new_models()
+
+            self._norm_models = {} if norm_models is None else norm_models
+            self._sign_models = {} if sign_models is None else sign_models
+            self.unit = None
+            self.stereo_combiner = StereoCombiner.from_name(
+                self.stereo_combiner_cls,
+                prefix=self.prefix,
+                property=self.property,
+                parent=self,
+            )
+        else:
+            loaded = self.read(self.load_path)
+            if (
+                subarray is not None
+                and loaded.subarray.telescope_types != subarray.telescope_types
+            ):
+                self.log.warning(
+                    "Supplied subarray has different telescopes than subarray loaded from file"
+                )
+            self.__dict__.update(loaded.__dict__)
+            self.subarray = subarray
+
+    def _new_models(self):
+        norm_regressor = SUPPORTED_REGRESSORS[self.norm_cls](**self.norm_config)
+        sign_classifier = SUPPORTED_CLASSIFIERS[self.sign_cls](**self.sign_config)
+        return norm_regressor, sign_classifier
+
+    def _table_to_y(self, table, mask=None):
+        """
+        Extract target values as numpy array from input table
+        """
+        # make sure we use the unit that was used during training
+        if self.unit is not None:
+            norm_y = np.abs(table[mask][self.target].quantity.to_value(self.unit))
+        else:
+            norm_y = np.abs(np.array(table[self.target][mask]))
+
+        if self.log_target:
+            if np.any(norm_y <= 0):
+                raise ValueError("norm_y contains negative values, cannot apply log")
+            norm_y = np.log(norm_y)
+
+        sign_y = np.sign(np.array(table[self.target][mask]))
+        return norm_y, sign_y
+
+    def fit(self, key, table):
+        """
+        Create and fit new models for ``key`` using the data in ``table``.
+        """
+        self._norm_models[key], self._sign_models[key] = self._new_models()
+
+        X, valid = table_to_X(table, self.features, self.log)
+        self.unit = table[self.target].unit
+        norm_y, sign_y = self._table_to_y(table, mask=valid)
```

Comment:
I would just call it `norm` and `sign`. `norm_y` is confusing...

Reply:
Done